src/main/java/polyglot/ServiceList.java

```diff
@@ -0,0 +1,108 @@
+package polyglot;
+
+import java.io.File;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Joiner;
+import com.google.protobuf.DescriptorProtos.FileDescriptorSet;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Descriptors.MethodDescriptor;
+import com.google.protobuf.Descriptors.ServiceDescriptor;
+
+import polyglot.protobuf.ServiceResolver;
+
+/** Simple utility to list the services, methods and message definitions for the known GRPC end-points */
+public class ServiceList {
+
+  /** Lists the set of GRPC services - optionally filtered by service name (contains) or method name (contains) */ 
+  public static void listServices(FileDescriptorSet fileDescriptorSet, String protoDiscoveryRoot,
+      Optional<String> serviceFilter, Optional<String> methodFilter, Optional<Boolean> withMessage) {
+
+    ServiceResolver serviceResolver = ServiceResolver.fromFileDescriptorSet(fileDescriptorSet);
+
+    // Add white-space before the rendered output
+    System.out.println();
```

Comment:
This isn't logging - it's functional output to stdout that you might subsequently want to pipe to another tool (e.g. grep).

My personal preference is to have logging on stderr (which it currently is), and output that is to be used by the user (either directly or indirectly) is written as "clean text" (i.e. with no logging meta data such as time etc.) to stdout. Happy to discuss - and ultimately you have the casting vote.